### PR TITLE
147177 - bess bs maintain cell edition in azure table

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
This PR consists change of [PBI 147177](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/147177)

- Code changes
- Unit test cases

### CODE REVIEW

**Review by shirinbt14926 — COMMENTED** [review]
Minor comment please

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Models/TableEntities/BessProductVersionEntities.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Azure;
+using Azure.Data.Tables;
+
+namespace UKHO.PeriodicOutputService.Common.Models.TableEntities
+{
+    [ExcludeFromCodeCoverage]
+    public class BessProductVersionEntities : ITableEntity
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Models/TableEntities/BessProductVersionEntities.cs:8]: Can we use existing ProductVersionEntities.cs class for it and if yes we can delete this as its almost same ?

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/UKHO.BESS.BuilderService.UnitTests.csproj [hunk 2]
```diff
@@ -8,6 +8,14 @@
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <NoWarn>1701;1702;8604;8618;8625</NoWarn>
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/UKHO.BESS.BuilderService.UnitTests.csproj:12]: What we are ignoring ?

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/UKHO.BESS.BuilderService.UnitTests.csproj (hunk 2 above)
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.BESS.BuilderService.UnitTests/UKHO.BESS.BuilderService.UnitTests.csproj]: Resolved

File: NVDSuppressions.xml [hunk 3]
```diff
@@ -249,4 +240,11 @@
     </notes>
     <cve>CVE-2022-39349</cve>
   </suppress>
+  <suppress>
+   <notes><![CDATA[
+   file name: Azure.Identity:1.11.0
+   Low level vulnerability. Package is set to latest version, when next fix is released must be updated to resolved CVE.
+   ]]></notes>
+   <vulnerabilityName>CVE-2024-29992</vulnerabilityName>
+</suppress>
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply NVDSuppressions.xml]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Models/TableEntities/BessProductVersionEntities.cs (hunk 1 above)
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Models/TableEntities/BessProductVersionEntities.cs]: Resolved

**Review by shirinbt14926 — APPROVED** [review]
LGTM. Thanks

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Pull request merged**